src/Avalonia.Visuals/Animation/Transitions/BrushTransition.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using Avalonia.Animation.Animators;
+using Avalonia.Media;
+
+namespace Avalonia.Animation
+{
+    /// <summary>
+    /// Transition class that handles <see cref="AvaloniaProperty"/> with <see cref="IBrush"/> type.
+    /// Only values of <see cref="ISolidColorBrush"/> will correctly transition.
+    /// </summary>
+    public class ISolidColorBrushTransition : Transition<IBrush>
```

Comment:
How would we add new transition that will work with other IBrush implementations? It seems to require breaking changes in the future.

Reply:
Though I doubt it will be added any time soon

Reply:
Depends, transitions are manual so we can add `GradientBrushTransition` and then users need to use it and it will break no compat.

Other option is to use `BrushTransition` but then again it does not match animator naming.

Reply:
I've renamed it to `BrushTransition` in the end, it will deal with not matching types and just do a discrete jump between values if we can't do a smooth transition.